nightshift/comms/sierra_search_platform.py

```diff
@@ -0,0 +1,337 @@
+# -*- coding: utf-8 -*-
+
+"""
+This module handles communication with NYPL Platform and BPL Solr.
+It is used to check status of records in Sierra between WorldCat queries
+so fully cataloged or deleted bibs are dropped from the process.
+"""
+import logging
+import os
+from typing import Optional
+
+from bookops_nypl_platform import PlatformToken, PlatformSession
+from bookops_nypl_platform.errors import BookopsPlatformError
+from bookops_bpl_solr import SolrSession
+from bookops_bpl_solr.session import BookopsSolrError
+from requests import Response
+
+from .. import __title__, __version__
+from ..ns_exceptions import SierraSearchPlatformError
+
+logger = logging.getLogger("nightshift")
+
+
+def is_eresource_callno(callno: str) -> bool:
+    """
+    Checks if call number is for electronic resource
+
+    Args:
+        callno:                         call number string
+
+    Returns:
+        bool
+    """
+    try:
+        norm_callno = callno.lower()
+    except AttributeError:
+        return False
+
+    if norm_callno.startswith("enypl"):  # NYPL pattern
+        return True
+    elif norm_callno in ("ebook", "eaudio", "evideo"):  # BPL pattern
+        return True
+    else:
+        return False
+
+
+class SearchResponse:
+    def __init__(self, sierraId: int, library: str, response: Response) -> None:
+        """
+        Initiates SearchResponse object.
+
+        Args:
+            sierraId:                   Sierra bib number
+            library:                    'nyp' or 'bpl'
+            response:                   `requests.Response` instance from the service
+
+        Raises:
+            `ns_exceptions.SierraSearchPlatformError`
+        """
+        self.sierraId = sierraId
+        self.library = library
+
+        if response.status_code > 404:
```

Comment:
Is it possible that this could ever get a `403` error for an authorization issue? Just trying to think of edge cases that might cause problems.

Reply:
I think it is possible. Thanks! Modified.

Reply:
I kept a separate treatment of NYPL Platform 404 errors but maybe I'm overthinking it.